src/gwt/src/org/rstudio/studio/client/workbench/views/connections/ConnectionsPresenter.java

```diff
@@ -248,123 +247,61 @@ public void onActiveConnectionsChanged(ActiveConnectionsChangedEvent event)
    {
       updateActiveConnections(event.getActiveConnections());
    }
+
+   private void terminateShinyApp()
```

Comment:
I have a mental todo on this one, @jmcphers and I talked about this, will merge and address this couple comments afterwards to start working in the same branch.